android/lib/src/main/java/com/swmansion/gesturehandler/NativeViewGestureHandler.java

```diff
@@ -86,7 +88,14 @@ protected void onHandle(MotionEvent event) {
         view.onTouchEvent(event);
         activate();
       } else if (state != STATE_BEGAN) {
-        begin();
+        // setting flag for an exclusive touch for buttons
+        if (view instanceof RNGestureHandlerButtonViewManager.ButtonViewGroup) {
```

Comment:
not sure I like adding this check here. I'd prefer to try one of the following ways:
1) Can we use `shouldRecognizeSimultaneously` abstraction to implement this behavior?
2) Can we add it directly to `NativeViewGestureHandler` or maybe even to all types of gesture handlers?

Reply:
1. not sure, since `shouldRecognizeSimultaneously` allows for staying in began state and I don't want to change this logic.

2. I'm not convinced how could it look like without any `instanceof RNGestureHandlerButtonViewManager.ButtonViewGroup`. I moved it to another function for a bit cleaner code.